libcoap/src/resource.rs

```diff
@@ -313,6 +314,30 @@ impl<D: Any + ?Sized + Debug> CoapResource<D> {
         }
     }
 
+    /// Adds an attribute with a name and an optional value to a CoAP resource object.
+    /// # Safety
+    /// libcoap uses the passed pointers "attr_name" and "attr_val" here.
+    /// Because of the passed flags, libcoap takes ownership of these pointers.
+    /// Passing NULL for the value is safe, since libcoap only frees the value, if it's not NULL.
+    pub fn add_attr(&self, name: &str, val: Option<&str>) {
+        let mut inner = self.inner.borrow_mut();
+        unsafe {
+            let attr_name = coap_new_str_const(name.as_ptr(), name.len());
+            let attr_val = match val {
+                Some(val_str) => coap_new_str_const(val_str.as_ptr(), val_str.len()),
+                None => std::ptr::null_mut(),
+            };
+            // coap_new_str_const allocates memory, which should be freed when
+            // the resource is deleted. Freeing a null pointer is a no-op here
+            coap_add_attr(
+                inner.raw_resource,
+                attr_name,
+                attr_val,
+                (COAP_ATTR_FLAGS_RELEASE_NAME | COAP_ATTR_FLAGS_RELEASE_VALUE) as i32,
+            );
+        }
+    }
+
```

Comment:
Also supporting `coap_find_attr()` would be a nice to have (but feel free to ignore this comment if it is non-trivial to implement it).

Reply:
Moved to new issue #46